[relnotes] Relnote update Marta - df106 tutorials (#13959)
root-project/root: README/ReleaseNotes/v630/index.md

 Will Buttinger, Rutherford Appleton Lab,\
 Philippe Canal, FNAL,\
 Olivier Couet, CERN/EP-SFT,\

>  Marta Czurylo, CERN/EP-SFT,\

Reviewer: YIKES sorry! Can you check that I didn't forget you in v6-30-00-patches (which is where we build the release notes from, i.e. the branch that actually matters for this change)? Please also backport this to v6-30-00-patches!

- The `RDatasetSpec` class and its users now employ the concept of 'sample' rather than the original naming 'group' for groups of files with associated metadata.

> - `df106_HiggsToFourLeptons` tutorials (both python and C++) now showcase addition of the ProgressBar, usage of the FromSpec function in order to define multiple samples and usage of the systematic variations with Vary function.

Reviewer: ```suggestion
- `df106_HiggsToFourLeptons` tutorials (both python and C++) now showcase the `ProgressBar`. They now use `FromSpec` to define multiple samples and `Vary` for systematic variations.
```
